compiler/semexprs.nim

```diff
@@ -1613,10 +1613,17 @@ proc semDefined(c: PContext, n: PNode, onlyCurrentScope: bool): PNode =
   # we replace this node by a 'true' or 'false' node:
   result = newIntNode(nkIntLit, 0)
   if not onlyCurrentScope and considerQuotedIdent(c.config, n[0], n).s == "defined":
-    if n.sons[1].kind != nkIdent:
-      localError(c.config, n.info, "obsolete usage of 'defined', use 'declared' instead")
-    elif isDefined(c.config, n.sons[1].ident.s):
-      result.intVal = 1
+    case n.sons[1].kind
```

Comment:
it's easier than this. Use:

```nim

let d = considerQuotedIdent(c.config, n[1], n)
result.intVal = isDefined(c.config, d.s)
```

Reply:
Oohhh, that's nice! So many helpful procs in the compiler :)